presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketing.java

```diff
@@ -321,49 +323,78 @@ private static int hashBytes(int initialValue, Slice bytes)
         return Optional.of(new HiveBucketFilter(builder.build()));
     }
 
-    private static OptionalInt getHiveBucket(Table table, Map<ColumnHandle, NullableValue> bindings)
+    private static Optional<Set<Integer>> getHiveBuckets(Table table, Map<ColumnHandle, Set<NullableValue>> bindings)
     {
-        if (bindings.isEmpty()) {
-            return OptionalInt.empty();
-        }
+        Optional<List<SimpleImmutableEntry<String, TypeInfo>>> nameToType = getBucketColumnTypeInfos(table);
```

Comment:
we tend not to use Pair-like objects; here, I suggest inlining getBucketColumnTypeInfos like so:

```
        List<String> bucketColumns = table.getStorage().getBucketProperty().get().getBucketedBy();
        if (bucketColumns.isEmpty()) {
            return Optional.empty();
        }

        Map<String, HiveType> hiveTypes = table.getDataColumns().stream()
                .collect(toImmutableMap(Column::getName, Column::getType));

        // Verify the bucket column types are supported
        for (String column : bucketColumns) {
            if (!SUPPORTED_TYPES_FOR_BUCKET_FILTER.contains(hiveTypes.get(column))) {
                return Optional.empty();
            }
        }

        Map<String, Set<NullableValue>> nameToBindings = bindings.entrySet().stream()
                .collect(toImmutableMap(entry -> ((HiveColumnHandle) entry.getKey()).getName(), Entry::getValue));

        ImmutableList.Builder<Set<NullableValue>> orderedBindingsBuilder = ImmutableList.builder();
        for (String columnName : bucketColumns) {
            if (!nameToBindings.containsKey(columnName)) {
                return Optional.empty();
            }
            orderedBindingsBuilder.add(nameToBindings.get(columnName));
        }

        List<Set<NullableValue>> orderedBindings = orderedBindingsBuilder.build();

        int bucketCount = table.getStorage().getBucketProperty().get().getBucketCount();
        List<TypeInfo> types = bucketColumns.stream()
                .map(hiveTypes::get)
                .map(HiveType::getTypeInfo)
                .collect(toImmutableList());

        ImmutableSet.Builder<Integer> buckets = ImmutableSet.builder();
        getHiveBuckets(new Object[types.size()], 0, orderedBindings, bucketCount, types, buckets);

        return Optional.of(buckets.build());
```

Reply:
Sure.